src/Microsoft.ServiceFabric.Services.Remoting/V2/RemoteException2.cs

```diff
@@ -0,0 +1,29 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.ServiceFabric.Services.Remoting.V2
+{
+    using System.Collections.Generic;
+    using System.Runtime.Serialization;
+
+    [DataContract(Name = "RemoteException2", Namespace = Constants.ServiceCommunicationNamespace)]
+    internal class RemoteException2
```

Comment:
Can we change the class name for one of the RemoteException classes. The existing RemoteException class and this class have same name and serve two very different purposes.

Reply:
We cannot change the older class because of compat issue. The newer one is anyways internal. I have added comment in the class summary to clarify the purpose of this class.